### PR TITLE
Python client changes for new PUT endpoints/ functionality

### PR DESCRIPTION
**Pull request by seanshi-scale**
Testing:
Ran this against locally-hosted Gateway and it seems to send the celery request correctly

### CODE REVIEW

File: nucleus/deploy/client.py [hunk 1]
```diff
@@ -426,6 +419,49 @@ def create_model_endpoint(
...
+        if gpus == 0 and gpu_type is not None:
+            logger.warning("GPU type setting %s will have no effect", gpu_type)
+            payload["gpu_type"] = None
+        for k, v in payload.copy().items():
+            if v is None:
+                del payload[k]
+        resp = self.connection.put(payload, f"{ENDPOINT_PATH}/{endpoint_name}")
+        endpoint_creation_task_id = resp.get("endpoint_creation_task_id", None)
+        logger.info(
+            "Endpoint creation task id is %s", endpoint_creation_task_id
```
Comment (yixu34) [inline nucleus/deploy/client.py:462]: Endpoint edit task id?

File: nucleus/deploy/client.py [hunk 2]
```diff
@@ -426,6 +419,49 @@ def create_model_endpoint(
...
             )
 
+    def edit_model_endpoint(
+        self,
+        endpoint_name: str,
+        model_bundle: Optional[ModelBundle] = None,
+        cpus: Optional[float] = None,
+        memory: Optional[str] = None,
+        gpus: Optional[int] = None,
+        min_workers: Optional[int] = None,
```
Comment (yixu34) [inline nucleus/deploy/client.py:429]: Maybe this is clearer in the server side PR (I've so far mainly been looking at the abstractions and less so the semantics 😅 ) but if someone omits a field, do you read the existing value and ensure that it's preserved?

**Review by yixu34 — APPROVED** [review]

File: nucleus/deploy/client.py (hunk 2 above)
Comment (seanshi-scale) [reply nucleus/deploy/client.py:429]: Yup, that's the intended behavior

**Comment by seanshi-scale** [comment]
We have our own client now, don't need this anymore

**Pull request closed without merging**